heron/executor/src/python/heron_executor.py

```diff
@@ -844,8 +844,11 @@ def _get_instance_plans(self, packing_plan, container_id):
       if container_plan.id == container_id:
         this_container_plan = container_plan
 
-    # make sure that our shard id is a valid one
-    assert this_container_plan is not None
+    # When the executor runs in newly added container by `heron update`,
+    # there is no plan for this container. In this situation,
+    # return None to bypass instance processes.
+    if this_container_plan is None:
+      return None
```

Comment:
currently, executor uses the list to start instances, I think. If we skip here, where is this function called again to get the list and start the instances?

Reply:
the new repacking plan triggers the executor to launch the actual instnaces

Reply:
kk. thx.